packages/engine/Specs/Scene/Model/ModelInstanceSpec.js

```diff
@@ -0,0 +1,417 @@
+              4073970.3334786133,
+            ),
+          ),
+        },
+        scene,
+      );
+
+      const boundingSphere = instance.getBoundingSphere(model);
+      expect(
+        ModelInstance.prototype.getPrimitiveBoundingSphere,
+      ).toHaveBeenCalled();
+
+      console.log("boundingSphere --> ", boundingSphere);
+
+      const boundingSphereCenter = new Cartesian3(
+        1253575.4296778704,
+        -4732902.902639246,
+        4073980.8573345104,
+      );
+      const boundingSphereRadius = 3.9121082161607323;
+
+      expect(boundingSphere.center).toEqual(boundingSphereCenter);
+      expect(boundingSphere.radius).toEqual(boundingSphereRadius);
+    });
+
+    it("computes bounding sphere for primitive in world space, earth centered model", async function () {
+      const translationPosition = new Cartesian3(
+        1253565.1903550967,
+        -4732913.141962121,
+        4073970.3334786133,
+      );
+      const headingPositionRoll = new HeadingPitchRoll();
+      const fixedFrameTransform = Transforms.localFrameToFixedFrameGenerator(
+        "north",
+        "west",
+      );
+      const instanceModelMatrix = new Transforms.headingPitchRollToFixedFrame(
+        translationPosition,
+        headingPositionRoll,
+        Ellipsoid.WGS84,
+        fixedFrameTransform,
+      );
+
+      const instance = new ModelInstance(instanceModelMatrix);
+
+      spyOn(ModelInstance.prototype, "computeModelMatrix").and.callThrough();
+
+      // values based on the Primitive for the "Wheels" Node in the CesiumMilkTruck
+      const sampleModel = {
+        modelMatrix: Matrix4.IDENTITY,
+        _sceneGraph: {
+          rootTransform: new Matrix4(
+            0,
+            0,
+            1,
+            0,
+            1,
+            0,
+            0,
+            0,
+            0,
+            1,
+            0,
+            0,
+            0,
+            0,
+            0,
+            1,
+          ),
+        },
+      };
+
+      const samplePrimitiveBoundingSphere = new BoundingSphere(
+        new Cartesian3(0, 0, 0),
+        1.2187656103276185,
+      );
+
+      const sampleRuntimeNode = {
+        computedTransform: new Matrix4(
+          -5.867126234708317e-8,
+          -0.999999916273327,
+          1.0506962941054424e-8,
+          -8.539378371944296e-8,
+          -0.17627758151258402,
+          0,
+          -0.9843404120540511,
+          0.4277219815419916,
+          0.9843404120540493,
+          -5.9604641561248813e-8,
+          -0.17627758151258371,
+          1.4326699118214847,
+          0,
+          0,
+          0,
+          1,
+        ),
+      };
+
+      const primitiveBoundingSphere = instance.getPrimitiveBoundingSphere(
+        sampleModel.modelMatrix,
+        sampleModel._sceneGraph,
+        sampleRuntimeNode,
+        samplePrimitiveBoundingSphere,
+      );
+
+      expect(ModelInstance.prototype.computeModelMatrix).toHaveBeenCalled();
```

Comment:
Do we need this expect case? I don't think the bounding sphere values will be correct without it, so we can assume it was called correctly.